pynamodb/models.py

```diff
@@ -975,3 +1030,23 @@ def _serialize_keys(cls, hash_key, range_key=None):
         if range_key is not None:
             range_key = cls._range_key_attribute().serialize(range_key)
         return hash_key, range_key
+
+
+class _ModelFuture:
+
+    def __init__(self, model_cls):
+        self._model_cls = model_cls
+        self._model = None
+        self._resolved = False
+
+    def update_with_raw_data(self, data):
+        if data:
+            self._model = self._model_cls.from_raw_data(data=data)
+        self._resolved = True
+
+    def get(self):
+        if not self._resolved:
+            raise Exception('Cannot resolve until request has been processed')
```

Comment:
```suggestion
            raise RuntimeException('Cannot get until resolved')
```
unless we want the language to be specifically about transactions, which might be OK-ish

Reply:
im fine with updating to be less specific